firebase-firestore/src/jsMain/kotlin/dev/gitlive/firebase/firestore/internal/NativeFirebaseFirestoreWrapper.kt

```diff
@@ -20,28 +21,41 @@ import kotlinx.coroutines.GlobalScope
 import kotlinx.coroutines.await
 import kotlinx.coroutines.promise
 
+// There is currently no way to check whether Firestore was already initialized for a given app without actually initializing it
+// Therefore we keep track of this internally
+private val appsWithFirestore = mutableListOf<FirebaseApp>()
+
 internal actual class NativeFirebaseFirestoreWrapper internal constructor(
     private val createNative: NativeFirebaseFirestoreWrapper.() -> NativeFirebaseFirestore,
+    private val canUpdateSettings: () -> Boolean,
 ) {
 
-    internal actual constructor(native: NativeFirebaseFirestore) : this({ native })
+    internal actual constructor(native: NativeFirebaseFirestore) : this({ native }, { false })
     internal constructor(app: FirebaseApp) : this(
         {
             NativeFirebaseFirestore(
-                initializeFirestore(app, settings.js).also {
-                    emulatorSettings?.run {
-                        connectFirestoreEmulator(it, host, port)
+                if (appsWithFirestore.contains(app)) {
+                    getFirestore(app)
+                } else {
+                    initializeFirestore(app, settings.js).also {
+                        emulatorSettings?.run {
+                            connectFirestoreEmulator(it, host, port)
+                        }
+                        appsWithFirestore.add(app)
                     }
                 },
             )
         },
+        {
+            !appsWithFirestore.contains(app)
+        },
     )
 
     private data class EmulatorSettings(val host: String, val port: Int)
 
     actual var settings: FirebaseFirestoreSettings = FirebaseFirestoreSettings.Builder().build()
         set(value) {
-            if (lazyNative.isInitialized()) {
+            if (lazyNative.isInitialized() || !canUpdateSettings()) {
                 throw IllegalStateException("FirebaseFirestore has already been started and its settings can no longer be changed. You can only call setFirestoreSettings() before calling any other methods on a FirebaseFirestore object.")
```

Comment:
I thought we removed this check in favour of letting the native sdk throw the exception?

Reply:
The thing is that there is no native sdk to fall back to, as explained in the other remark. We shouldnt be calling initializeFirestore here yet, as settings may still be updated after this.